api/client.go

```diff
@@ -510,15 +510,24 @@ func (c *Config) ReadEnvironment() error {
 	}
 
 	if envVaultProxy != "" {
-		u, err := url.Parse(envVaultProxy)
+		err := c.SetTransportProxy(envVaultProxy)
 		if err != nil {
 			return err
 		}
+	}
+
+	return nil
+}
 
-		transport := c.HttpClient.Transport.(*http.Transport)
-		transport.Proxy = http.ProxyURL(u)
+func (c *Config) SetTransportProxy(proxy string) error {
```

Comment:
I think this is incorrect - this is for Vault Proxy, not a Proxy in a traditional sense (i.e. nginx). The new `-proxy-address` field should work identically to the `-agent-address` field. The old `-agent-address` field was for Vault Agent's API Proxy (which later got moved into Vault Proxy's API Proxy). The word 'proxy' is a little overloaded, so I think this is where the confusion came from.

But yeah, tl;dr: handle `-proxy-address` the same as we do `-agent-address`

Reply:
Ah, so if I'm understanding you correctly, when `-proxy-address` is passed on the CLI, we're just wanting to set the value of `config.AgentAddress` with the value, the same way we do with `-agent-address`. `-proxy-address` is effectively just an alias for `-agent-address`?

I did it this way because I was looking at the client code [here](https://github.com/hashicorp/vault/blob/main/api/client.go#L460-L463) and [here](https://github.com/hashicorp/vault/blob/main/api/client.go#L512-L520) and trying to emulate what happens when the `VAULT_PROXY_ADDR` env var is set.

Reply:
Oh, huh. You're totally right. It looks like I confused myself on the docs. I think I'd assumed that `VAULT_PROXY_ADDR` was the Proxy version of `VAULT_AGENT_ADDR` once things moved to Proxy, and that was why the `AGENT` one wasn't documented. Instead, it looks like the `AGENT` one was never documented?

So, I do think that you have faithfully made a version of `VAULT_PROXY_ADDR`, but that leaves us in a weird spot (no fault of yours, to be clear), since we have two undocumented variables for Agent that should be pointing to Proxy that don't have an equivalent.

I think this probably is correct, then, and we should merge this. It's not what I thought this command was, but it is an improvement.

If we wanted to add an equivalent for `-agent-address` and `VAULT_AGENT_ADDR` for Vault Proxy in the future, we could use `-api-proxy-address` and `VAULT_API_PROXY_ADDR` which would probably be much clearer anyway. There's no burning need since it's not actually documented.

Reply:
Sorry for the confusion here -- like I said, I think I had assumed we'd changed over the docs from `VAULT_AGENT_ADDR` to `VAULT_PROXY_ADDR` when we made the switch, but it looks like we just hadn't had the former documented to begin with. It does leave us an interesting question for the future but I think it's one that's easy enough to solve.